base/sparse/sparsematrix.jl

```diff
@@ -1390,7 +1390,77 @@ function one{T}(S::SparseMatrixCSC{T})
+end
+"""
+Returns a `SparseMatrixCSC` that shares `A`'s structure but with stored entries
+`broadcast(f, A.nzval)`.
+"""
+function _broadcast_zeropres{Tf}(f::Tf, A::SparseMatrixCSC)
+    Bcolptr = copy(A.colptr); resize!(Bcolptr, A.n + 1)
+    Browval = copy(A.rowval); resize!(Browval, nnz(A))
+    Bnzval = broadcast(f, A.nzval); resize!(Bnzval, nnz(A))
+    return SparseMatrixCSC(A.m, A.n, Bcolptr, Browval, Bnzval)
+    # TODO: The following lines for constructing Bcolptr and Browval might be safer than
+    # those above: If A.colptr and/or A.rowval are shorter than their expected lengths, the
+    # copy!s might catch the error, whereas the former copy/resize! combinations won't. The
+    # lines above are simpler/cleaner though. Thoughts?
+    # Bcolptr = similar(A.colptr, A.n + 1)
+    # copy!(Bcolptr, 1, A.colptr, 1, A.n + 1)
+    # Browval = similar(A.rowval, nnz(A))
+    # copy!(Browval, 1, A.rowval, 1, nnz(A))
+end
+"""
+Returns a (dense) `SparseMatrixCSC` with `f(zero(eltype(A)))` stored in place of all unstored
+entries in `A` and `broadcast(f, A.nzval)` stored in place of all stored entries in `A`.
+"""
+function _broadcast_notzeropres{Tf}(f::Tf, A::SparseMatrixCSC)
+    nnzB = A.m * A.n
+    # Build structure
+    Bcolptr = similar(A.colptr, A.n + 1)
+    copy!(Bcolptr, 1:A.m:(nnzB + 1))
+    Browval = similar(A.rowval, nnzB)
+    for k in 1:A.m:(nnzB - A.m + 1)
+        copy!(Browval, k, 1:A.m)
+    end
+    # Populate values
+    Bnzval = fill(f(zero(eltype(A))), nnzB)
+    @inbounds for (j, jo) in zip(1:A.n, 0:A.m:(nnzB - 1)), k in nzrange(A, j)
+        Bnzval[jo + A.rowval[k]] = f(A.nzval[k])
+    end
+    # NOTE: Combining the fill call into the loop above to avoid multiple sweeps over /
+    # nonsequential access of Bnzval does not appear to improve performance
+    return SparseMatrixCSC(A.m, A.n, Bcolptr, Browval, Bnzval)
+end
+
+# Cover common broadcast operations involving a single sparse matrix and one or more
+# broadcast scalars.
+#
+# TODO: The minimal snippet below is not satisfying: A better solution would achieve
+# the same for (1) all broadcast scalar types (Base.Broadcast.containertype(x) == Any?) and
+# (2) any combination (number, order, type mixture) of broadcast scalars. One might achieve
+# (2) via a generated function. But how one might simultaneously and gracefully achieve (1)
+# eludes me. Thoughts much appreciated.
+#
+broadcast{Tf}(f::Tf, x::Union{Number,Bool}, A::SparseMatrixCSC) = broadcast(y -> f(x, y), A)
+broadcast{Tf}(f::Tf, A::SparseMatrixCSC, y::Union{Number,Bool}) = broadcast(x -> f(x, y), A)
+# NOTE: The following two method definitions work around #19096. These definitions should
+# be folded into the two preceding definitions on resolution of #19096.
+broadcast{Tf,T}(f::Tf, ::Type{T}, A::SparseMatrixCSC) = broadcast(y -> f(T, y), A)
+broadcast{Tf,T}(f::Tf, A::SparseMatrixCSC, ::Type{T}) = broadcast(x -> f(x, T), A)
```

Comment:
The hacky snippet below takes a different tack. In principle it should handle any combination of broadcast scalars and `SparseMatrixCSC`s; in practice type inference fails when the number of arguments exceeds four or more than one `SparseMatrixCSC` is involved.
```julia
import Base: tail, front
import Base.Broadcast: broadcast_c, containertype, promote_containertype, broadcast_indices

broadcast_indices(::Type{SparseMatrixCSC}, A) = indices(A)
containertype{T<:SparseMatrixCSC}(::Type{T}) = SparseMatrixCSC
# Combinations of sparse arrays with broadcast scalars should yield sparse arrays
promote_containertype(::Type{Any}, ::Type{SparseMatrixCSC}) = SparseMatrixCSC
promote_containertype(::Type{SparseMatrixCSC}, ::Type{Any}) = SparseMatrixCSC
# Combinations of sparse arrays with anything else should fall back to dense array methods
promote_containertype(::Type{Array}, ::Type{SparseMatrixCSC}) = Array
promote_containertype(::Type{SparseMatrixCSC}, ::Type{Array}) = Array
promote_containertype(::Type{Tuple}, ::Type{SparseMatrixCSC}) = Array
promote_containertype(::Type{SparseMatrixCSC}, ::Type{Tuple}) = Array

broadcast_c(f, ::Type{SparseMatrixCSC}, As...) = foo((x, y) -> f(y...), (), first(As), tail(As))

foo(f, arrays, nextarg, remargs) =
    foo((x, y) -> f(x, (nextarg, y...)), arrays, first(remargs), tail(remargs))
foo(f, arrays, nextarg::SparseMatrixCSC, remargs) =
    foo((x, y) -> f(front(x), (last(x), y...)), (arrays..., nextarg), first(remargs), tail(remargs))

foo(f, arrays, nextarg, remargs::Tuple{}) =
    bar(x -> f(x, (nextarg,)), arrays...)
foo(f, arrays, nextarg::SparseMatrixCSC, remargs::Tuple{}) =
    bar(x -> f(front(x), (last(x),)), arrays..., nextarg)

bar(f, A) = broadcast(x -> f((x,)), A)
bar(f, A, B) = broadcast_c((x,y) -> f((x,y)), Array, A, B)
bar(f, A, B, Cs...) = broadcast_c((x,y,zs...) -> f((x,y,zs...)), Array, A, B, Cs...)
```
I'm sure a better approach exists, possibly e.g. inverting the call chain. Having a look at `@pabloferz`'s suggestion now. Best!